propagators/gcp/src/main/java/com/google/cloud/opentelemetry/propagators/XCloudTraceContextPropagator.java

```diff
@@ -0,0 +1,131 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.propagators;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceId;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A context propagator that leverages the X-Cloud-Trace-Context header.
+ *
+ * <p>See: <a href="https://cloud.google.com/trace/docs/setup#force-trace">Google Cloud Trace
+ * Documentation</a> for details.
+ */
+public final class XCloudTraceContextPropagator implements TextMapPropagator {
+
+  private static String FIELD = "x-cloud-trace-context";
+  private static Collection<String> FIELDS = Collections.singletonList(FIELD);
+  private static Pattern VALUE_PATTERN =
+      Pattern.compile("(?<traceid>[0-9a-f]{32})\\/(?<spanid>[\\d]{1,20});o=(?<sampled>\\d+)");
+  private static Logger LOGGER = Logger.getLogger("XCloudTraceContextPropogator");
+
+  private final boolean oneway;
+
+  /**
+   * Constructs a new text map propogator that leverages the X-Cloud-Trace-Context header.
+   *
+   * @param oneway
+   */
+  public XCloudTraceContextPropagator(boolean oneway) {
+    this.oneway = oneway;
+  }
+
+  @Override
+  public Collection<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, C carrier, TextMapSetter<C> setter) {
+    // We never inject cloud trace context information in oneway mode.
+    if (oneway) {
+      return;
+    }
+    Span current = Span.fromContext(context);
+    if (!current.getSpanContext().isValid()) {
+      return;
+    }
+    String sampledString = current.getSpanContext().isSampled() ? "1" : "0";
+    String spanIdString =
+        java.lang.Long.toUnsignedString(
+            java.lang.Long.parseUnsignedLong(current.getSpanContext().getSpanId(), 16));
+    String value =
+        current.getSpanContext().getTraceId() + "/" + spanIdString + ";o=" + sampledString;
+    setter.set(carrier, FIELD, value);
+  }
+
+  @Override
+  public <C> Context extract(Context context, C carrier, TextMapGetter<C> getter) {
+    if (context == null || getter == null) {
+      return context;
+    }
+    // Ignore the scenario where another propagator already filled out a field.
+    if (Span.fromContext(context).getSpanContext().isRemote()) {
```

Comment:
do we want to support the scenario in which someone wants to prefer XCTC over traceparent?  I think this prevents that usage.

Reply:
It doesn't, because you can order the propagators in the java autoconfigure (or manual configure) module.